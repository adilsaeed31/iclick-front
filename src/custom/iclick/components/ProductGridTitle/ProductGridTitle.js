import React, { Component } from "react";
import PropTypes from "prop-types";

export default class ProductGridTitle extends Component {
  static propTypes = {
    displayTitle: PropTypes.string
  };

  static defaultProps = {
    displayTitle: ""
  };

  render() {
    const { displayTitle } = this.props;

    if (!displayTitle) return null;

    return <h2>{displayTitle}</h2>;
  }
}
