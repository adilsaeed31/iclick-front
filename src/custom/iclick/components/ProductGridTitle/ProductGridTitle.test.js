import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "mobx-react";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "custom/reactionTheme";
import ProductGridTitle from "./ProductGridTitle";

const uiStore = {
  appConfig: {
    publicRuntimeConfig: {
      placeholderImageUrls: {
        productGrid: ""
      }
    }
  }
};

test("snapshot of a tag with a displayTitle", () => {
  const component = renderer.create(<MuiThemeProvider theme={theme}>
    <Provider uiStore={uiStore}>
      <ProductGridTitle displayTitle={"Test Title"} />
    </Provider>
  </MuiThemeProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshop of tag without a displayTitle", () => {
  const component = renderer.create(<MuiThemeProvider theme={theme}>
    <Provider uiStore={uiStore}>
      <ProductGridTitle displayTitle={""} />
    </Provider>
  </MuiThemeProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
