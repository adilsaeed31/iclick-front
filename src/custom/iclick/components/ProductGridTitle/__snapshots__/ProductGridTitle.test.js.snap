// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshop of tag without a displayTitle 1`] = `null`;

exports[`snapshot of a tag with a displayTitle 1`] = `
<h1
  className="MuiTypography-root-2 MuiTypography-title-8 SkProductGridTitle-gridTitle-1"
>
  Test Title
</h1>
`;
